scripts/packages/VSCodeServer/src/progress.jl

```diff
@@ -0,0 +1,27 @@
+struct VSCodeLogger <: Logging.AbstractLogger end
+
+function Logging.handle_message(j::VSCodeLogger, level, message, _module,
+                                group, id, file, line; kwargs...)
+    progress = ProgressLogging.asprogress(level, message, _module, group, id, file, line; kwargs...)
```

Comment:
Is this the only thing we are using from `ProgressLogging`? Essentially a function that takes a message from the logging system and puts it into a struct? Might it make sense to actually copy that code into here and then no longer ship `ProgressLogging` at all? Really the interface we are relying is the format of the log message, not some types in this package, right?

In general it looks to me like ProgressLogging.jl is the package that packages should use that want to report progress, so maybe it would be nice to move the stuff that a `AbstractLogger` needs out of that package?

Reply:
Hm, not sure. I just pushed a commit that simply copies the relevant code from ProgressLogging.jl here, but it turns out that that's a fairly substantial part of the package.
That said, it makes https://github.com/JunoLab/ProgressLogging.jl/pull/34 unnecessary and it's probably a good thing to keep those "hacks" here...

Reply:
> Really the interface we are relying is the format of the log message, not some types in this package, right?

No, quite the opposite. The intent is that the `Progress` type is the definitive "type for reporting progress", and that the other pattern matching which is done in `asprogress` is a hack to support the older `progress` keyword.